File: python3/add_digits.py
"""
Space   : O(1)
Time    : O(n)
"""


class Solution:
    def addDigits(self, num: int) -> int:
        if num < 10:
            return num

        while len(str(num)) > 1:
            n = str(num)
            temp = 0
            for i in n:
                temp += int(i)
            num = temp

        return num

File: golang/add_digits.go
package main

import (
	"fmt"
)

func addDigits(num int) int {
	for true {
		num_str := fmt.Sprint(num)
		num = 0
		for _, c := range num_str {
			num += int(c) - 48
		}

		num_str = fmt.Sprint(num)
		if len(num_str) == 1 {
			return num
		}
	}
	return num
}
